### PR TITLE
Introduce docker build bits for mixer.

### PR DESCRIPTION
**Pull request by douglas-reid**
At the moment, building a docker image for mixer appears to only work on
linux machines. As such, these builds should only be conducted on linux
hosts.

This setup should closely align with the manager build images and push
images to the same istio-test repo.

This attempts to begin to address #162.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/186)

### CODE REVIEW

File: docker/build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -ex
+
+gcloud config set project istio-test
```
Comment (mandarjog) [inline docker/build.sh]: we should rename the file  gcloud_build.sh.
It is important to have a script that just works in our environment.

Comment (douglas-reid) [inline docker/build.sh]: OK. This bit just duplicates the docker/build.sh from istio/manager. I'll rename to gcloud_build.sh. Ultimately, we just need a place to put our images.

**Comment by douglas-reid** [comment]
PTAL.

Moved: `docker/build.sh` to `docker/gcloud_build.sh`

File: BUILD.debian [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# Extract .xz files
+genrule(
+    name = "jessie_tar",
+    srcs = ["docker-brew-debian-b39d31635ca26c8b1f3d982090ba8d54167c4d85/jessie/rootfs.tar.xz"],
+    outs = ["jessie_tar.tar"],
+    cmd = "cat $< | xzcat >$@",
+)
+
+docker_build(
+    name = "jessie",
```
Comment (douglas-reid) [inline BUILD.debian]: does envoy require ubuntu (and trusty in particular)? seems like we might want xenial (if we stick with ubuntu).

just looking at the image sizes via pantheon, the envoy and manager images seem to be significantly larger than the mixer image (~60-70 vs 50 MBs). I was already considering moving mixer to jessie-slim to decrease the image size. not sure how important that will become.

Comment (kyessenov) [inline BUILD.debian]: I tried making Envoy with public debian xenial image but there was mismatch with GLIBC dependencies, and I did not proceed. For better or worse, it's ubuntu trusty for now. The thing with docker is that if we all share base layer, docker will only download it once.

**Comment by douglas-reid** [comment]
PTAL. Updated to xenial as the base image.

**Pull request merged**